httpcore/_async/http2.py

```diff
@@ -346,4 +362,5 @@ async def body_iter(self, timeout: TimeoutDict) -> AsyncIterator[bytes]:
                 break
 
     async def _response_closed(self) -> None:
+        self.connection.max_streams_semaphore.release()
```

Comment:
I'm thinking testing this one is tricky in our current high-level testing policy in `test_interfaces` as we need quite a lot of requests to be performed to an external server which seems a bit much.

Here in httpcore we don't have a `server` fixture, we could introduce one but it'd have to support HTTP/2 to cover this case.

Reply:
Indeed. We've at least got test *coverage* over the lines, which is a good first pass, even if we're not testing all kinds of behaviour.

Reply:
Right, it was more of an overall comment on our testing strategy moving forwards, not that much to do with this particular PR. We'll address separately 👍